src/main/java/club/sk1er/mods/autogg/handlers/gg/AutoGGHandler.java

```diff
@@ -0,0 +1,111 @@
+package club.sk1er.mods.autogg.handlers.gg;
+
+import club.sk1er.mods.autogg.AutoGG;
+import club.sk1er.mods.autogg.handlers.patterns.PatternHandler;
+import club.sk1er.mods.autogg.tasks.data.Server;
+import club.sk1er.mods.autogg.tasks.data.Trigger;
+import club.sk1er.mods.autogg.tasks.data.TriggerType;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.entity.EntityPlayerSP;
+import net.minecraftforge.client.event.ClientChatReceivedEvent;
+import net.minecraftforge.event.entity.EntityJoinWorldEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.modcore.api.utils.Multithreading;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Where the magic happens...
+ * We handle which server's triggers should be used
+ * and how to detect which server the player is currently
+ * on.
+ *
+ * @author ChachyDev
+ */
+
+public class AutoGGHandler {
+    private volatile Server server;
+
+    @SubscribeEvent
+    public void onEntityJoinWorld(EntityJoinWorldEvent event) {
```

Comment:
use `WorldEvent.Load` here instead, this is going to fire for every single player

Reply:
wouldn't exactly work with WorldEvent.Load without very "hacky" code.

Reply:
then change it to `event.entity == Minecraft.getMinecraft.thePlayer`